File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-3.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int len = 0;
        ListNode curr = head;
        while(curr!=null) {
            len++;
            curr = curr.next;
        }
        if(len==n) return head.next;
        curr = head;
        len = len-n;
        while(len > 1) {
            curr = curr.next;
            len--;
        }
        curr.next = curr.next.next;
        return head;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-3.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0,l1)
        prev = dummy
        carry = 0
        while l1 or l2 or carry!=0:
            a = l1.val if l1 else 0
            b = l2.val if l2 else 0
            sum = a+b+carry
            carry = sum//10
            sum = sum%10
            if l1:
                l1.val = sum
                prev = l1
                l1 = l1.next
            else:
                prev.next = ListNode(sum)
                prev = prev.next
            if l2:
                l2 = l2.next
        return dummy.next
